kernel: remove symbols found in generic

neheb opened the pull request:
Found with basic fish loop.

for f in (find target -name "config-6.6" | grep -v generic)
  if grep -qFx -f target/linux/generic/config-6.6 $f
    grep -vF -f target/linux/generic/config-6.6 $f > tmpfile && mv tmpfile $f
  end
end

ping @hauke @Ansuel

neheb committed:
kernel: filter out compiler opts from config

These get dynamically set based on compiler version. Not relevant for
targets.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16770
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

KanjiMonster commented on target/linux/generic/config-6.6:
```diff
@@ -834,6 +834,8 @@ CONFIG_CARDBUS=y
 # CONFIG_CC10001_ADC is not set
 # CONFIG_CCS811 is not set
 CONFIG_CC_CAN_LINK=y
+CONFIG_CC_IMPLICIT_FALLTHROUGH="-Wimplicit-fallthrough=5"
+CONFIG_CC_NO_ARRAY_BOUNDS=y
```
All these `CONFIG_(G)CC_*` values (apart from `CC_OPTIMIZE_FOR_*`) are dynamically set based on the used compiler (version), and should be moved to https://github.com/openwrt/openwrt/blob/main/target/linux/generic/config-filter so they don't show up even if their values may change due to switching compiler versions.

neheb replied:
Done. Fixed up my other commit too.

neheb commented:
That is such a fascinating failure...

hauke commented on target/linux/armsr/armv7/config-6.6:
```diff
@@ -1,7 +1,6 @@
 CONFIG_ALIGNMENT_TRAP=y
 CONFIG_ARCH_32BIT_OFF_T=y
 CONFIG_ARCH_MIGHT_HAVE_PC_PARPORT=y
-CONFIG_ARCH_MMAP_RND_BITS=8
```
This is needed because the target base configuration sets it to 18, see:
```
target/linux/armsr/armv7/config-6.6:CONFIG_ARCH_MMAP_RND_BITS=8
target/linux/armsr/armv8/config-6.6:CONFIG_ARCH_MMAP_RND_BITS=18
target/linux/armsr/config-6.6:CONFIG_ARCH_MMAP_RND_BITS=18
```

neheb replied:
sounds broken. config-6.6 has CONFIG_64BIT but armv7 is a 32-bit platform.

hauke replied:
armsr/armv7/config-6.6 overwrites the settings from armsr/config-6.6 again

hauke commented:
The first commit looks fine, the second one needs very detailed review. Could you split this PR up into two PRs please.

hauke commented on target/linux/airoha/an7581/config-6.6:
```diff
@@ -232,7 +230,6 @@ CONFIG_FUNCTION_ALIGNMENT_4B=y
 CONFIG_FWNODE_MDIO=y
 CONFIG_FW_CACHE=y
 # CONFIG_FW_LOADER_USER_HELPER is not set
-CONFIG_GCC10_NO_ARRAY_BOUNDS=y
```
Please remove `GCC11_NO_ARRAY_BOUNDS` too from all files.
`GCC11_NO_ARRAY_BOUNDS` was renamed to `CONFIG_GCC10_NO_ARRAY_BOUNDS` in https://github.com/gregkh/linux/commit/a5b60c8b9a9e602c83f9e7a714c690cb9acdba61

hauke replied:
Maybe add this as a separate commit

neheb replied:
Done.

neheb committed:
kernel: remove GCC11_NO_ARRAY_BOUNDS

This symbol is no longer present.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16770
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented on target/linux/generic/config-filter:
```diff
@@ -3,8 +3,11 @@
+# CONFIG_CC_NO_ARRAY_BOUNDS is not set
 CONFIG_CLANG_VERSION=.*
 CONFIG_FRAME_WARN=.*
+# CONFIG_GCC10_NO_ARRAY_BOUNDS is not set
```
Shouldn't it be?:
CONFIG_CC_NO_ARRAY_BOUNDS=*
CONFIG_GCC10_NO_ARRAY_BOUNDS=.*
